### ISSUE TITLE
Dependency Library + CLI

### ISSUE THREAD

**Issue by awgreene** [operator-framework/operator-controller#206]
We want to be able to perform deppy resolution off cluster. The Deppy library leaves it to users to define the EntitySource/VariableSources, so it wouldn't make sense to couple the CLI with the Deppy library. Instead, this ticket proposes that we create a new library that:
- Handles the implementation details of OLM V1's EntitySource/VariableSource implementation.
- Defines an input API that is OLM V1 specific:
-- FBC Content
-- Operator CR Constraints
-- Cluster Properties
- Introduces a CLI that allows off cluster resolution

A stretch goal could be to provide a KUBECONFIG and generate a file that can be used as an input API to the CLI.

**Comment by m1kola** [comment]
We have a proof of concept in this PR: https://github.com/operator-framework/operator-controller/pull/246

The next steps here:
* Add a message into the CLI tool saying something along the lines: "This is the proof of concept of off-cluster resolution and will likely change. We are gathering community feedback here: [link to GitHub discussion]"
* Get it reviewed and merged
* Gather more feedback from the community using this GitHub discussion - https://github.com/operator-framework/operator-controller/discussions/262
* Plan next iterations based on feedback

Also there are a few things going in parallel. For example, we will likely change significantly how we build resolution problems (see https://github.com/operator-framework/deppy/issues/96 and [relevant RFC](https://docs.google.com/document/d/1PizF1uQcMrrAFsLgcx2O0Coq7mei4xe9hxpz-QRp-x0/edit#heading=h.x3tfh25grvnv)). Once we have more or less stable way to build the resolution problem - we will be able to create some sort of library to share between operator-controller and resolution CLI.

**Comment by m1kola** [comment]
#246 got merged. We have PoC CLI which is:
* Capable of resolving off-cluster (doesn't require a cluster at all)
* Takes package name, channel (optional) and version (optional) as input for resulution
* It also take either a local directory with FBC or an image ref with FBC catalog (just one for now)
* Optionally it also takes a directory with yaml manifests. If manifests contain `Operator` objects, they will be used as input into the solver (list of required packages - similar to Operator objects installed in the cluster when running on-cluster resoltion).

Next steps:
* Gather feedback from the community using this GitHub discussion - https://github.com/operator-framework/operator-controller/discussions/262
* Plan next iterations based on feedback


> Also there are a few things going in parallel. For example, we will likely change significantly how we build resolution problems (see [operator-framework/deppy#96](https://github.com/operator-framework/deppy/issues/96) and [relevant RFC](https://docs.google.com/document/d/1PizF1uQcMrrAFsLgcx2O0Coq7mei4xe9hxpz-QRp-x0/edit#heading=h.x3tfh25grvnv)). Once we have more or less stable way to build the resolution problem - we will be able to create some sort of library to share between operator-controller and resolution CLI.

This is still valid.

**Comment by m1kola** [comment]
Created a new issue for next steps: #327
